### ISSUE TITLE
ci image breaking with old datadir for [typed_store::rocks: error=(de)serialization error: invalid value: integer `2`]

### ISSUE THREAD

**Issue by jun0tpyrc** [MystenLabs/sui#16843]
mysten/sui-node:5e74783cbb4881d1e4c0a53befddb6a446a6445e still good
newer ones from  roughly 23Mar  (not sure which commit  , for example 1.20.1-d951895b90 ) is breaking

```
2024-03-25T00:03:16.462681Z  INFO sui_core::authority::authority_store: Loading epoch start config from DB
2024-03-25T00:03:16.473722Z ERROR typed_store::rocks: error=(de)serialization error: invalid value: integer `2`, expected variant index 0 <= i < 2
2024-03-25T00:03:16.474569Z ERROR sui_node: Failed to start node: Storage error: (de)serialization error: invalid value: integer `2`, expected variant index 0 <= i < 2
```

**Comment by ebmifa** [comment]
@stefan-mysten where exactly are you seeing this?

**Comment by stefan-mysten** [comment]
@jun0tpyrc we need more information from you regarding where is this crashing. 
- commit
- branch / release branch you used
- share a gist with the complete logs?

**Comment by jun0tpyrc** [comment]
looks it resolved in auto in image 1.21.1-5f2bfb15ea, confirmed with two nodes

**Comment by stefan-mysten** [comment]
@jun0tpyrc Glad to hear that, and thanks for updating us.
